### DISCUSSION TITLE
Optimize Next/Image where layout=fill

### DISCUSSION THREAD

**Discussion by RyKilleen** [vercel/next.js#23646] [Help] [answered]
Hi all, first of all `next/image` is awesome.

I have a use-case where I don't always know the aspect ratio of an image in a card header. I have it set to

```
<Image
        src={image}
        layout="fill"
        objectFit="fill"
      />
```

I _know_ this image has a max-width of 500px, lets say, because the card has a max-width. I will never know it's height, because editors upload different ratio images all the time.

With the auto-generated srcset produced by next.config, is there any way for me to override and indicate that the largest image should be 500? Right now I always receive the largest screen-size version available, ie the `3840w` version.

Worth noting that it's passing this value along to Cloudinary, ie `<image_url>/f_auto,c_limit,w_3840`

Thanks for any help understanding the right way forward here!

**Comment by akellbl4** [comment, marked as answer]
Hi!
You can use `objectFit="cover"` and set both `width` and `height` and an image will be scaled one of the sizes. You can set a big number of pixels for height and an image will always be scaled by width.

**Reply by RyKilleen** [reply]
Just wanted to follow up and say that

```
layout="intrinsic"
width={800}
height={450}
objectFit="cover"
```

Does precisely what I need, as you recommended. This isn't covered in the docs (and in fact the docs seem to indicate this is not supported.)

**Reply by ragnsan** [reply]
But what if you need to use layout="fill" for a background image in the hero section? I have this exact problem. Anyone have a fix for this? I also get the w_3840 for all screen sizes

**Reply by Kyoss79** [reply]
@ragnsan uses sizes. 
e.g.
sizes="(max-width: 400px) 400px, 800px"

**Reply by ragnsan** [reply]
Hmm. I have noticed that none  of my images load different versions according to the screen size. I am using SSG and Cloudinary for my hosting. Could this be causing the issue? When I try to load an image locally, I get the intended behaviour, but when I use Cloudinary as a loader / provider, it only serves the 3840px version. Anyone have any knowledge on this topic?
